modules/proposals/discussion/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+//! Proposals discussion module for the Joystream platform. Version 2.
+//! Contains discussion subsystem for the proposals engine.
+//!
+//! Supported extrinsics:
+//! - add_post - adds a post to existing discussion thread
+//!
+//! Public API:
+//! - create_discussion - creates a discussion
+//!
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// Do not delete! Cannot be uncommented by default, because of Parity decl_module! issue.
+//#![warn(missing_docs)]
+
+#[cfg(test)]
+mod tests;
+mod types;
+
+use rstd::clone::Clone;
+use rstd::prelude::*;
+use rstd::vec::Vec;
+use runtime_primitives::traits::EnsureOrigin;
+use srml_support::{decl_module, decl_storage, Parameter};
+
+use types::{Post, Thread};
+
+//TODO: create_thread() ensures
+//TODO: create_post() ensures
+//TODO: select storage container for the posts (double map, inside thread)?
+//TODO: events?
+
+/// 'Proposal discussion' substrate module Trait
+pub trait Trait: system::Trait {
+    /// Origin from which thread author must come.
+    type ThreadAuthorOrigin: EnsureOrigin<Self::Origin, Success = Self::AccountId>;
+
+    /// Origin from which commenter must come.
+    type PostAuthorOrigin: EnsureOrigin<Self::Origin, Success = Self::AccountId>;
+
+    /// Discussion thread Id type
+    type ThreadId: From<u32> + Into<u32> + Parameter + Default + Copy;
+
+    /// Post Id type
+    type PostId: From<u32> + Parameter + Default + Copy;
+
+    /// Type for the thread author id. Should be authenticated by account id.
+    type ThreadAuthorId: From<Self::AccountId> + Parameter + Default;
```

Comment:
There should not be a `From<AccountId>` here, there need not be any injective mapping from Account space to `ThreadAuthorId`.  E.g. two members can have the same role account at any given time. Currently, this is actually not possible, but we will allow this in the future:

https://github.com/Joystream/substrate-runtime-joystream/issues/93

Reply:
Answered in general comment below.

Reply:
Ok.